pkg/proxyserver/log/logClient.go

```diff
@@ -0,0 +1,117 @@
+package log
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	workManagerLogSAName = "klusterlet-addon-workmgr-log"
+)
+
+type Handler struct {
+	KubeClient                           kubernetes.Interface
+	BaseDomain                           string
+	CaData                               []byte
+	ClusterName                          string
+	PodNamespace, PodName, ContainerName string
+	Query                                url.Values
+}
+
+func (c *Handler) logClient() (kubernetes.Interface, error) {
+	logTokenSecret, err := c.KubeClient.CoreV1().Secrets(c.ClusterName).Get(context.Background(),
+		workManagerLogSAName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	userServerHost := "cluster-proxy-user." + c.BaseDomain
+
+	clusterProxyCfg := &rest.Config{
+		Host: fmt.Sprintf("https://%s/%s", userServerHost, c.ClusterName),
```

Comment:
I think we should directly use grpc client for cluster-proxy instead of using rest service. You can check how it works in clusteradm.